homeassistant/components/roomba/vacuum.py

```diff
@@ -135,6 +78,20 @@ def __init__(self, name, roomba):
         self.vacuum = roomba
         self.vacuum_state = None
 
+    @property
+    def unique_id(self):
+        """Return the uniqueid of the vacuum cleaner."""
+        return self._name
+
+    @property
+    def device_info(self):
+        """Return the device info of the vacuum cleaner."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": "iRobots",
+            "name": str(self._name),
```

Comment:
```
"mobilityVer":"7375","bootloaderVer":"36","soundVer":"13"
```
It looks like the api gives us back version data which would allow a "sw_version" field
https://developers.home-assistant.io/docs/device_registry_index/

Reply:
If it is useful for a followup PR if you add a sensor to track the battery % ```"batteryType":"lith","batPct":100``` it would show attached to the device since this is now a config entry!
<img width="1238" alt="Screen Shot 2020-03-27 at 11 01 37 AM" src="https://user-images.githubusercontent.com/663432/77775322-64279e00-701a-11ea-9948-d41d8ee36a22.png">

Reply:
It looks like `mac` is provided as well, which would allow other integrations like unifi to tell when the device is offline

```
  "connections": {(CONNECTION_NETWORK_MAC, self.mac)},
```

Reply:
I added a sensor for the battery and the trash if the data exists